storage/storage.c

```diff
@@ -838,6 +868,20 @@ static secbool decrypt_dek(const uint8_t *kek, const uint8_t *keiv) {
     return secfalse;
   }
 
+  uint8_t kek[SHA256_DIGEST_LENGTH] = {0};
+  uint8_t keiv[SHA256_DIGEST_LENGTH] = {0};
+  if (get_lock_version() >= 5) {
+    if (sectrue !=
+        derive_kek_unlock(pin, pin_len, storage_salt, ext_salt, kek)) {
```

Comment:
What is the reason for calling `derive_kek_unlock` from `decrypt_dek` instead of `unlock`? In my opinion, it reduces readability.

Is the reason to ensure that EDEK is in storage before verifying the PIN?